cleverhans/loss.py

```diff
@@ -0,0 +1,50 @@
+import json
+import os
+import tensorflow as tf
+
+from .model import Model
+
+
+class Loss(object):
+    """
+    An abstract interface for loss wrappers that allows flexible control of
+    real examples, adversarial examples and labels..
+    """
+
+    def __init__(self, model, hparams, attack=None):
```

Comment:
It might be worth thinking about having a similar style to the interface we have for the Model and Attack class when it comes to the optional parameters (hparams here)? What do you think?

Reply:
Yes possibly, in my case I refrained from modifying existing APIs. But for future PRs that could help to have coherent style.

Reply:
That's fine, we can clean up and make the entire API more homogeneous in a later PR if needed